### PR TITLE
Feature/arrays

### PR DESCRIPTION
**Pull request by Tanane**

### CODE REVIEW

File: staff/khaoula-tanane/arrays/array-functions/index-of.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+function indexOf(array, string, start=0){
```
Comment (lichab) [inline staff/khaoula-tanane/arrays/array-functions/index-of.js:1]: So here, you are using a ES6 feature and at the momento that is not allowed. You just have to replicate the behaviour inside the function :)

File: staff/khaoula-tanane/arrays/array-functions/reduce.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+function reduce(array, callback, initialValue = 0){
```
Comment (lichab) [inline staff/khaoula-tanane/arrays/array-functions/reduce.js:1]: same here

File: staff/khaoula-tanane/arrays/array-functions/includes.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+function includes(array, element, start = 0){
```
Comment (lichab) [inline staff/khaoula-tanane/arrays/array-functions/includes.js:1]: same here

File: staff/khaoula-tanane/arrays/arroz/arroz.prototype.includes.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Array.prototype.includes = function(element, start = 0){
```
Comment (lichab) [inline staff/khaoula-tanane/arrays/arroz/arroz.prototype.includes.js:1]: same here

File: staff/khaoula-tanane/arrays/array-functions/index-of.js (hunk 1 above)
Comment (lichab) [reply staff/khaoula-tanane/arrays/array-functions/index-of.js:1]: Kaula you marked this as resolved but it was not modified

File: staff/khaoula-tanane/arrays/array-functions/includes.js (hunk 3 above)
Comment (lichab) [reply staff/khaoula-tanane/arrays/array-functions/includes.js:1]: Don't mark things as resolved if no changes were made

Comment (Tanane) [reply staff/khaoula-tanane/arrays/array-functions/includes.js:1]: > Don't mark things as resolved if no changes were made

I just changed it this morning! :)

File: staff/khaoula-tanane/arrays/arroz/arroz.prototype.index-of.js [hunk 5]
```diff
@@ -1,4 +1,5 @@
-Arroz.prototype.indexOf = function(string, start=0){
+Arroz.prototype.indexOf = function(string, start){
+    var start = 0
```
Comment (lichab) [inline staff/khaoula-tanane/arrays/arroz/arroz.prototype.index-of.js:2]: Watch out here, you are making start to **always** be 0. And it should be 0 **only** if no value is provided.

Comment (Tanane) [reply staff/khaoula-tanane/arrays/arroz/arroz.prototype.index-of.js]: Thank you!  I did not realize that it was giving me an error.

File: staff/khaoula-tanane/arrays/arroz/arroz.prototype.slice.js [hunk 6]
```diff
@@ -2,6 +2,8 @@
 
 Arroz.prototype.slice = function(start, end){
 
+    
+
```
Comment (CaliZam) [inline staff/khaoula-tanane/arrays/arroz/arroz.prototype.slice.js:6]: You hi consider if start its negative value. However, nice code

**Comment by lichab** [comment]
Hay Kaula, great job in general! All the code looks really good, I only left you some comments on some best practices to take into accunt. 

Unfortunately I will have to close the PR because it has changes in the precourse folder, and as we already explained each branch can only have changes on the folders that belong to it. Even if that is only renaming files. 

To fix this you have to follow the steps on the guide found at `stuff/git/README` => ***How to correct a wrongly made Pull Request*** 

Let me know if you need help with the procedure

**Pull request closed without merging**